src/scripts/main.js

```diff
@@ -1,3 +1,129 @@
 'use strict';
 
-// write your code here
+function keyListener(e) {
+  if (game.getStatus() !== 'playing') {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowDown':
+      game.moveDown();
+      break;
+    case 'ArrowLeft':
+      game.moveLeft();
+      break;
+    case 'ArrowRight':
+      game.moveRight();
+      break;
+    case 'ArrowUp':
+      game.moveUp();
+      break;
+  }
+
+  update();
+}
+
+function update() {
+  game.getState().forEach((row, i) => {
+    row.forEach((el, j) => {
+      let cellCssClass = 'field-cell';
+
+      cellCssClass += el >= 2 ? ` field-cell--${el}` : '';
+      gameCells[i][j].innerText = el || '';
+      gameCells[i][j].className = cellCssClass;
+    });
+  });
+
+  scoreElement.innerText = game.getScore();
+
+  switch (game.getStatus().toUpperCase()) {
+    case 'WIN':
+      changeMessage('WIN');
+      break;
+    case 'LOSE':
+      changeMessage('LOSE');
+      break;
+    case 'NO_MOVES':
+      changeMessage('NO_MOVES');
+      break;
+    default:
+      hideAllMessages();
+      break;
+  }
+}
+
+function start() {
+  switch (game.getStatus()) {
```

Comment:
The 'start' function handles multiple game statuses and controls UI state. It would be beneficial to break this function down into smaller, more focused functions to improve readability and maintainability.